quix-frontend/client/src/lib/sql-autocomplete/sql-context-evaluator/tree-analyzer.ts

```diff
@@ -0,0 +1,107 @@
+import { QueryDetails, TableInfo, TableType } from './types';
+import { createNewTableInfoObj } from './utils';
+
+export const analyzeRelation = (relationNode: any): TableInfo[] => {
+  if (relationNode.relation) {
+    // when use 'JOIN' keyword -> analyze all relations
+    return relationNode
+      .relation()
+      ?.reduce((accumulator: TableInfo[], subRelation: any) => {
+        return accumulator.concat(...analyzeRelation(subRelation));
+      }, []);
+  }
+
+  const aliasedRelationNode = relationNode.sampledRelation().aliasedRelation();
+  const relationPrimaryNode = aliasedRelationNode.relationPrimary();
+
+  if (relationPrimaryNode.relation) {
+    return analyzeRelation(relationPrimaryNode.relation());
+  }
+
+  const currentTableInfo: TableInfo = createNewTableInfoObj({
+    name:
+      relationPrimaryNode.qualifiedName &&
+      relationPrimaryNode.qualifiedName()?.getText(),
+    type: relationPrimaryNode.query ? TableType.Nested : TableType.External,
+    alias:
+      aliasedRelationNode.identifier &&
+      aliasedRelationNode.identifier()?.getText(),
+  });
+
+  if (currentTableInfo.type === TableType.Nested && relationPrimaryNode.query) {
+    const nestedQueryDetails: QueryDetails = analyzeQuerySpecificationNode(
+      getNextQuerySpecificationNode(relationNode)
+    );
+    aggregateQueryAndTableInfo(nestedQueryDetails, currentTableInfo);
+  }
+
+  return [currentTableInfo];
```

Comment:
there's a bit of disconnect between what the function returns and how it's named.
I suppose it can be called something like `getRelationInfo`.